DragaliaAPI.Test/Unit/Controllers/DungeonRecordControllerTest.cs

```diff
@@ -0,0 +1,579 @@
+                        DungeonType: DungeonTypes.None,
+                        Scene01: "",
+                        AreaName01: "",
+                        Scene02: "",
+                        AreaName02: "",
+                        Scene03: "",
+                        AreaName03: "",
+                        Scene04: "",
+                        AreaName04: "",
+                        Scene05: "",
+                        AreaName05: "",
+                        Scene06: "",
+                        AreaName06: ""
+                    )
+                }
+            );
+
+        this.mockQuestRepository
+            .Setup(x => x.GetQuests(DeviceAccountId))
+            .Returns(new List<DbQuest>()
+                {
+                    new() { DeviceAccountId = DeviceAccountId, QuestId = questId }
+                }.AsQueryable().BuildMock());
+
+        this.mockUserDataRepository
+            .Setup(x => x.AddTutorialFlag(DeviceAccountId, 1022))
+            .ReturnsAsync(new DbPlayerUserData { });
+
+        this.mockQuestRepository
+            .Setup(x => x.CompleteQuest(DeviceAccountId, questId, It.IsAny<float>()))
+            .ReturnsAsync(
+                new DbQuest()
+                {
+                    DeviceAccountId = DeviceAccountId,
+                    QuestId = questId,
+                    IsMissionClear1 = true,
+                    IsMissionClear2 = true,
+                    IsMissionClear3 = true
+                }
+            );
+
+        this.mockUserDataRepository
+            .Setup(x => x.GetUserData(DeviceAccountId))
+            .Returns(new List<DbPlayerUserData>()
+                {
+                    new()
+                    {
+                        DeviceAccountId = DeviceAccountId,
+                        Name = "Euden",
+                        ViewerId = 1
+                    }
+                }.AsQueryable().BuildMock());
+
+        this.mockInventoryRepository
+            .Setup(
+                x => x.AddMaterials(DeviceAccountId, It.IsAny<List<Materials>>(), It.IsAny<int>())
+            )
+            .Returns(Task.FromResult(0));
+
+        this.mockUpdateDataService
+            .Setup(x => x.GetUpdateDataList(DeviceAccountId))
+            .Returns(new UpdateDataList() { });
+
+        this.mockQuestRepository.Setup(x => x.SaveChangesAsync()).ReturnsAsync(0);
+    }
+
+    // Tests that CompleteQuest returning the same clear time as GetQuests marks time as best clear time
+    [Fact]
+    public async Task BestClearDisplaysTimeAsBestTime()
+    {
+        this.mockQuestRepository
+            .Setup(x => x.GetQuests(DeviceAccountId))
+            .Returns(new List<DbQuest>()
+                {
+                    new()
+                    {
+                        DeviceAccountId = DeviceAccountId,
+                        QuestId = questId,
+                        BestClearTime = clearTime
+                    }
+                }.AsQueryable().BuildMock());
```

Comment:
You already have a setup for almost this exact same thing in the constructor
```csharp
 this.mockQuestRepository
            .Setup(x => x.GetQuests(DeviceAccountId))
            .Returns(new List<DbQuest>()
                {
                    new() { DeviceAccountId = DeviceAccountId, QuestId = questId }
                }.AsQueryable().BuildMock());
```

It's personal preference more than anything but I like to avoid putting setups in the constructor and keep them in the test methods, so that it's clear reading the method what I'm testing. Unless it's boilerplate stuff and not really relevant (I would say FinishDungeon falls under that umbrella). It's up to you, but it would probably be a good idea to remove the duplicate ones at least.